### ISSUE TITLE
Invalid test names for repeated and parametrized tests

### ISSUE THREAD

**Issue by olga999999** [mannodermaus/android-junit5#317]
Information about the test's parameter and repetition is not included in the test name when tests are executed not as a whole test class/suite, but separately, one by one. This leads to the automatically generated logcat logs having the same filename and therefore being overwritten on each repetition/parameter.
E.g. on the screenshot below the whole test suite is executed. Test names contain the information about repetitions and parameters.
<img width="435" alt="image" src="https://github.com/mannodermaus/android-junit5/assets/141128175/b6986335-bb5f-4159-89bb-8e4e1ef1a10c">
Here only one test is executed at a time. No information about repetitions and parameters.
<img width="406" alt="image" src="https://github.com/mannodermaus/android-junit5/assets/141128175/ec08d8da-00e3-42dc-9217-d05d903ae225">
<img width="380" alt="image" src="https://github.com/mannodermaus/android-junit5/assets/141128175/22672ed3-14c1-4b02-a3be-c1f4208e0568">

**Comment by mannodermaus** [comment]
Hey, thanks for reporting. I can't confirm that this is what I see on my end, the output is identical whether I execute the entire instrumentation test suite or a single class. 🤔 

|Single|Suite|
|---|---|
|<img width="600" alt="single" src="https://github.com/mannodermaus/android-junit5/assets/3897705/b10d1017-6a7f-4e3e-909f-e39022fab2bd">|<img width="600" alt="suite" src="https://github.com/mannodermaus/android-junit5/assets/3897705/d138d3db-01f1-4605-bfd0-8dda3fdf2f1b">|

May I ask what version of JUnit 5, the android-junit5 plugin and its instrumentation libraries you are using in your project? It's possible that a recent unreleased change has improved something that hasn't been publicized yet.

**Comment by mannodermaus** [comment]
Thank you for the quick response and explanation! No worries, I was the one who read it wrong – "only one test" was referring to a single test method, not a single test class. With this cleared up, I can reproduce this on my end as well:

<img width="424" alt="Screenshot 2024-04-10 at 9 16 24" src="https://github.com/mannodermaus/android-junit5/assets/3897705/5b63713f-6cbb-4893-bfd1-fe3ca2530097">

Having identical test names could create other problems down the line, so this is definitely something that needs fixing. Thanks for reporting!

**Comment by mannodermaus** [comment]
Good news! I narrowed down the cause for this to some [special case handling](https://github.com/mannodermaus/android-junit5/blob/main/instrumentation/runner/src/main/kotlin/de/mannodermaus/junit5/internal/runners/AndroidJUnitPlatformTestTree.kt#L74-L80) inside the test tree generator. Unfortunately, I cannot simply remove this safeguard, as it would break other single-method executions from the IDE (such as #199 and #207, which were the original motivation for adding this). The reason is a limitation in how the Android instrumentation expects its classic tests to be named. This restriction doesn't seem to apply to all kinds of tests though, so I may have an approach in the works. Stay tuned for more.

**Comment by mannodermaus** [comment]
This will be released alongside the upcoming instrumentation test libraries 1.5.0! If you need the fix now, you can use the `1.5.0-SNAPSHOT` in the meantime. 🙇‍♂️

**Comment by mannodermaus** [comment]
Not on Maven Central anyway, that's correct. You'd need to grab them from their snapshots repository by adding the following repository to your project:

```
repositories {
  maven { setUrl("https://oss.sonatype.org/content/repositories/snapshots/") }
}
```

Sorry, I usually add this to my comments when resolving tickets, but forgot this time. 🙈
